### PR TITLE
Synchronize pleno-qc to latest changes

### PR DESCRIPTION
**Pull request by antony-gigapixel-imaging**
Upgrade `pleno-qc` to the latest edition for `qcuser` in the Jupyter server. Ensure `pleno-common` is the most recent version.

Also download openhtf to the `qcuser`'s home folder. Don't install, don't use it for now. Keep it as a fallback option.

### CODE REVIEW

**Review by bo-at-pleno — APPROVED** [review]
Does someone need to actually run the ansible here? or are you setting up a CI/CD to kick it off automagically after merge?

File: roles/jupyterhub/tasks/qcuser.yml [hunk 1]
```diff
@@ -77,11 +77,21 @@
...
+    dest: "{{ qcuser.home }}/openhtf"
+  become: true
+  become_user: qcuser
+
 # Provides visualization code
 - name: Clone pleno-qc.git
   ansible.builtin.git:
     repo: git@github.com:Pleno-Inc/pleno-qc.git
-    version: optical-qc-engine
+    version: bo/optical-qc
```
Comment (bo-at-pleno) [inline roles/jupyterhub/tasks/qcuser.yml:94]: hmm, is this a git branch? can we not use this? i think it's not ready yet.

Comment (antony-gigapixel-imaging) [reply roles/jupyterhub/tasks/qcuser.yml:94]: I deleted the branch `optical-qc-engine`, but I can bring it back. Please advice.

Comment (bo-at-pleno) [reply roles/jupyterhub/tasks/qcuser.yml:94]: if you want let's fork a branch called `dev`  and target that for internal release. I'll periodically merge, `bo/optical-qc` is unstable, i push even when there's syntax error and things are broken

**Comment by antony-gigapixel-imaging** [comment]
> Does someone need to actually run the ansible here? 

Nobody except me. To avoid modifying the servers by mistakes, I also removed `hosts.ini` representing the target server IP addresses. Currently, running ansible will return "hosts and credentials not yet defined" message.

> or are you setting up a CI/CD to kick it off automagically after merge?

No. I am running it manually. the CI/CD only has one linter running. 

CI/CD auto kickoff won't work eitherway. Our setup requires SUDO most of the time.

File: roles/jupyterhub/tasks/qcuser.yml [hunk 2]
```diff
@@ -77,11 +77,21 @@
   become: true
   become_user: qcuser
 
+# Download the OpenHTF framework source code, but don't install it. We don't use
+# the framework's web frontend and the test history serialization with protobuf.
+- name: Clone google/openhtf
```
[the remark this answers is not in the document]
Comment (antony-gigapixel-imaging) [reply roles/jupyterhub/tasks/qcuser.yml:82]: FAANG companies would have called the action "eating my own dogfood", or "putting money where the mouth is". But yeah, I get what you mean on the company scale. :)

File: roles/jupyterhub/tasks/qcuser.yml (hunk 1 above)
Comment (antony-gigapixel-imaging) [reply roles/jupyterhub/tasks/qcuser.yml]: Using git SHA to lock `pleno-qc` version. Closing this thread.

**Pull request merged**